app/models/enterprise_role.rb

```diff
@@ -6,4 +6,22 @@ class EnterpriseRole < ActiveRecord::Base
   validates_uniqueness_of :enterprise_id, scope: :user_id, message: I18n.t(:enterprise_role_uniqueness_error)
 
   scope :by_user_email, joins(:user).order('spree_users.email ASC')
+
+  def self.receives_notifications_for(enterprise_id)
+    EnterpriseRole.where(enterprise_id: enterprise_id, receives_notifications: true).first
+  end
+
+  def self.set_notification_user(user_id, enterprise_id)
+    managers_for(enterprise_id).map do |m|
+      if m.user_id == user_id.to_i
+        m.update_attributes receives_notifications: true
+      elsif m.user_id != user_id.to_i && m.receives_notifications
+        m.update_attributes receives_notifications: false
+      end
+    end
+  end
+
+  def self.managers_for(enterprise_id)
+    EnterpriseRole.where(enterprise_id: enterprise_id)
+  end
```

Comment:
IMO this logic clearly fits into a service object so that we don't add yet another responsibility to the model. Besides, it's very clear to me what this `.set_notification_user` actually does :thinking:

Reply:
So the `EnterpriseRoles` model handles which users have permissions to manage which enterprises. We've changed the way enterprise emails get sent, so it's now a flag on a manager (EnterpriseRole). `.set_notification_user` sets the flag for which user out of the list of managers of an enterprise will be the one to receive the emails for orders etc.

I added a couple of methods to:
- return a list of all managers for an enterprise
- return the user that currently receives emails for a specific enterprise
- set the a user that will receive emails for an enterprise

Reply:
I am a bit confused by the need for all of this, I think it should all be on `Enterprise` (or a service object):
 
1) It seems like `EnterpriseRole.receives_notifications_for(enterprise_id)` and `enterprise.contact` are defined kind of backwards. Wouldn't it be simpler to do:

````ruby
# enterprise.rb
def contact
  contact = users.where(enterprise_roles: { receives_notifications: true} ).first
  contact || owner
end
````
Then you can ditch `receives_notifications_for` completely.

2) `EnterpriseRole.set_notification_user` can be rewritten on `Enterprise` as well:

````ruby
# enterprise.rb
def set_notification_user(user_id)
  enterprise_roles.update_all(["receives_notifications=(user_id=?)",user_id])
end
````

3) The above makes `EnterpriseRole.managers_for(enterprise_id)` redundant.